sdk/src/reserved_account_keys.rs

```diff
@@ -0,0 +1,202 @@
+//! Collection of reserved account keys that cannot be write-locked by transactions.
+//! New reserved account keys may be added as long as they specify a feature
+//! gate that transitions the key into read-only at an epoch boundary.
+
+#![cfg(feature = "full")]
+
+use {
+    crate::{
+        address_lookup_table, bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable,
+        compute_budget, config, ed25519_program, feature,
+        feature_set::{self, FeatureSet},
+        loader_v4, native_loader,
+        pubkey::Pubkey,
+        secp256k1_program, stake, system_program, sysvar, vote,
+    },
+    lazy_static::lazy_static,
+    std::collections::HashSet,
+};
+
+// Temporary until a zk token program module is added to the sdk
+mod zk_token_proof_program {
+    solana_sdk::declare_id!("ZkTokenProof1111111111111111111111111111111");
+}
+
+pub struct ReservedAccountKeys;
+impl ReservedAccountKeys {
+    /// Compute a set of all reserved keys, regardless of if they are active or not
+    pub fn active_and_inactive() -> HashSet<Pubkey> {
+        RESERVED_ACCOUNT_KEYS
+            .iter()
+            .map(|reserved_key| reserved_key.key)
+            .collect()
```

Comment:
how are these used? if we instead used a `HashMap` (probably in conjunction with the above change), can we move whatever tests these outputs are used for internally and directly query the map instead of allocating?

Reply:
It's only allocated once per bank. What are the keys and values of your proposed HashMap and what does it give us over a HashSet?